core/src/worker/workflow/mod.rs

```diff
@@ -1363,17 +1363,19 @@ fn prepare_to_ship_activation(wfa: &mut WorkflowActivation) {
         }
         fn variant_ordinal(v: &workflow_activation_job::Variant) -> u8 {
             match v {
-                workflow_activation_job::Variant::NotifyHasPatch(_) => 1,
-                workflow_activation_job::Variant::SignalWorkflow(_) => 2,
-                workflow_activation_job::Variant::DoUpdate(_) => 2,
+                workflow_activation_job::Variant::NotifyHasPatch(_) => 0,
+                workflow_activation_job::Variant::StartWorkflow(_) => 1,
+                workflow_activation_job::Variant::UpdateRandomSeed(_) => 2,
+                workflow_activation_job::Variant::SignalWorkflow(_) => 3,
+                workflow_activation_job::Variant::DoUpdate(_) => 3,
```

Comment:
I don't think this works. I think signal/update in the same task as start always need to come before start. It's important for data loss reason that signal/update handlers run before the primary method.

Reply:
Lengthy discussion outside of the PR summarized:

This change's impetus is to make the sending of _information_ be appropriately ordered, and lang can always assume that a main routine ought to be started at the end of the first activation.